nullaway/src/main/java/com/uber/nullaway/fixserialization/Serializer.java

```diff
@@ -213,9 +219,84 @@ public static String serializeSymbol(@Nullable Symbol symbol) {
         return symbol.name.toString();
       case METHOD:
       case CONSTRUCTOR:
-        return symbol.toString();
+        return serializeMethodSignature((Symbol.MethodSymbol) symbol);
       default:
         return symbol.flatName().toString();
     }
   }
+
+  /**
+   * Serializes the signature of the given {@link Symbol.MethodSymbol} to a string.
+   *
+   * @param methodSymbol The method symbol to serialize.
+   * @return The serialized method symbol.
+   */
+  private static String serializeMethodSignature(Symbol.MethodSymbol methodSymbol) {
+    if (methodSymbol == null) {
+      return "null";
+    }
+    StringBuilder sb = new StringBuilder();
+    if (methodSymbol.isConstructor()) {
+      Symbol.ClassSymbol encClass = methodSymbol.owner.enclClass();
+      // For constructors, we use the name of the enclosing class.
+      Name name = encClass.getSimpleName();
+      if (name.isEmpty()) {
+        // For anonymous classes, we use the name of the superclass.
+        name = encClass.getSuperclass().asElement().getSimpleName();
+      }
+      sb.append(name);
+    } else {
+      // For methods, we use the name of the method.
+      sb.append(methodSymbol.getSimpleName());
+    }
+    sb.append(
+        methodSymbol.getParameters().stream()
+            .map(
+                parameter ->
+                    parameter.type.accept(
+                        new Types.DefaultTypeVisitor<String, Void>() {
+                          @Override
+                          public String visitWildcardType(Type.WildcardType t, Void unused) {
+                            StringBuilder sb = new StringBuilder();
+                            sb.append(t.kind);
+                            if (t.kind != BoundKind.UNBOUND) {
+                              sb.append(t.type.accept(this, null));
+                            }
+                            return sb.toString();
+                          }
+
+                          @Override
+                          public String visitCapturedType(Type.CapturedType t, Void s) {
+                            return t.wildcard.accept(this, null);
+                          }
```

Comment:
If we get rid of serializing type arguments, I don't think we will need either of these two overrides either.  Maybe we can delete them?

Reply:
Yes exactly, since we decided to remove type arguments, we can implement this with simple iteration rather than a visitor. 424bdb5cc98a87d335d76571df5410762f08ce0a